File: Module6/PL_5.py
import sys
sys.path.insert(0, '/Users/evanedelstein/Desktop/School/2021/Spring2021/BTM-6000/Mods')
from Biostats import *
from sklearn.metrics import roc_curve
from sklearn.metrics import roc_auc_score


# df = pd.read_csv("/Users/evanedelstein/Desktop/School/2021/Spring2021/BTM-6000/Module6/Module 6 - NHANES subset.csv")
# print(df.head())
# # stage_1 =  df[(df["eGFR"] >= 90) & (df[]) ]
# stage_1 =  df[(df["eGFR"] >= 90) ]
# stage_2 =  df[(df["eGFR"] < 90) & (df["eGFR"] >= 60) ]
# stage_3=  df[(df["eGFR"] < 60) & (df["eGFR"] >= 30) ]
# stage_4 =  df[(df["eGFR"] < 30) & (df["eGFR"] >= 15) ]
# stage_1 =  df[df["eGFR"] < 15 ]

# def plot_roc_curve(fpr, tpr):
#     plt.plot(fpr, tpr, color='orange', label='ROC')
#     plt.plot([0, 1], [0, 1], color='darkblue', linestyle='--')
#     plt.xlabel('False Positive Rate')
#     plt.ylabel('True Positive Rate')
#     plt.title('Receiver Operating Characteristic (ROC) Curve')
#     plt.legend()
#     plt.show()


# print(dbinom(1,.13,11))


print(pbinom(1,11,0.13))

File: Module6/PL_5.R
df <- read.csv("/Users/evanedelstein/Desktop/School/2021/Spring2021/BTM-6000/Module6/Module 6 - NHANES subset.csv")

ls(df)
# Stages
df$CKD3_new <- ifelse(df$eGFR < 60,1,0 )

# 
df$MALE <- ifelse(df$gender ==1,1,0)
df$SCr_elevated_new <- NA
df$SCr_elevated_new[df$MALE ==1] <- ifelse(df$creatinine[df$MALE ==1] > 1.2,1,0 )
df$SCr_elevated_new[df$MALE ==0] <- ifelse(df$creatinine[df$MALE ==0] > 1.1,1,0 )
#2 contigencey table  2x2 
df <- df[which(!is.na(df$SCr_elevated_new)),]
TP<-length(df$SCr_elevated_new[df$SCr_elevated_new==1 & df$CKD3_new == 1])
FP<-length(df$SCr_elevated_new[df$SCr_elevated_new==1 & df$CKD3_new == 0])
TN<-length(df$SCr_elevated_new[df$SCr_elevated_new==0 & df$CKD3_new == 0])
FN<-length(df$SCr_elevated_new[df$SCr_elevated_new==0 & df$CKD3_new == 1])

df$SCr_elevated_factor<-factor(df$SCr_elevated,labels=c("SCr elevated","SCr not elevated"),levels=1:0)
df$CKD_progressed<-factor(df$CKD3,labels=c("CKD3 or more","no CKD to CKD 2"),levels=1:0)
table1 <- table(df$SCr_elevated_factor,df$CKD_progressed)
Sensitivity<-table1[1,1] / (table1[2,1]+table1[1,1] )
Specificity<-table1[2,2] / (table1[2,2]+table1[1,2] )
PPV<-table1[1,1] / (table1[1,2]+table1[1,1] )
NPV<-table1[2,2] / (table1[2,2]+table1[2,1] )
Sensitivity
Specificity
PPV
NPV
#ROC CURVE 
library(pROC)
roc1 <- roc(df$CKD_progressed,df$creatinine,plot=TRUE,grid=TRUE,print.auc=TRUE,ci=TRUE, show.thres=TRUE)
ls(roc1)
roc1$auc
roc1$ci
roc1$sensitivities
roc1$specificities


